test/integration/concierge_whoami_test.go

```diff
@@ -376,6 +364,33 @@ func TestWhoAmI_CSR_Parallel(t *testing.T) {
 	)
 }
 
+func maybeNeedsExtraWithSHA256(
+	t *testing.T,
+	discoveryClient discovery.DiscoveryInterface,
+	certPEM []byte,
+) map[string]identityv1alpha1.ExtraValue {
+	t.Helper()
+	var extra map[string]identityv1alpha1.ExtraValue
+
+	if testutil.KubeServerMinorVersionAtLeastInclusive(t, discoveryClient, 32) {
+		t.Log("Adding extra that includes SHA256 of the DER of the first cert")
+		block, _ := pem.Decode(certPEM)
+		require.NotEmpty(t, block)
+
+		bytesToHexOfSHA256 := func(b []byte) string {
+			shasum := sha256.Sum256(b)
+			return hex.EncodeToString(shasum[:])
+		}
+
+		sha256OfDER := fmt.Sprintf("X509SHA256=%s", bytesToHexOfSHA256(block.Bytes))
+
+		extra = map[string]identityv1alpha1.ExtraValue{
+			"authentication.kubernetes.io/credential-id": []string{sha256OfDER},
```

Comment:
See https://github.com/kubernetes/kubernetes/pull/125634 for the explanation of why this new field exists